base/rational.jl

```diff
@@ -226,6 +226,8 @@ signbit(x::Rational) = signbit(x.num)
 copysign(x::Rational, y::Real) = copysign(x.num,y) // x.den
 copysign(x::Rational, y::Rational) = copysign(x.num,y.num) // x.den
 
+abs(x::Rational{T}) where {T<:Unsigned} = x
```

Comment:
An alternative approach (and maybe more general?) would be to define
```
abs(x::Rational) = Rational(abs(x.num), x.den)
```
but I don't know what is best.
I also wonder if the `ifelse` in `numbers.jl` could simply be a ternary, i.e.
```
abs(x::Real) = signbit(x) ? -x : x
```
since, as noted in this PR, not all numbers can be negated.

Reply:
Indeed @fredrikekre's proposal appears to be more general. Should I add another commit to the same PR making the change or resubmit a cleaner one?

By the way, I was using unsigned rationals since it is a possible tag type for EXIF metadata. I agree that they are a strange choice in a general use case.

Reply:
Just add more commits here. Much easier that way. We can squash it when it’s ready.